pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -365,6 +375,17 @@ func (s *DiscoveryServer) removeCon(conID string, con *XdsConnection) {
 		log.Errorf("ADS: Removing connection for non-existing node %v.", s)
 	}
 	delete(adsClients, conID)
+	if _, ok := adsSidecarIDConnectionsMap[con.modelNode.ID]; ok {
```

Comment:
Don't particularly like this, should the slice instead be a `map[conID]*XdsConnection`?

Reply:
That was the edge case I told you about